### PR TITLE
More cleanup of API, introduces EnergyPacket and partial delivery

### PR DESCRIPTION
**Pull request by falkreon**
Let's discuss flyweight Packets and partial delivery!

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -43,4 +43,6 @@ dependencies {
 
 	//The main cotton api. We need a maven repo...
 	modCompile "com.github.CottonMC:cotton:master-SNAPSHOT"
+	
+	modCompile "com.google.code.findbugs:jsr305:3.0.2"
```
Comment (LemmaEOF) [inline build.gradle:47]: Does this need to stick around? I don't see any uses of it in the code at the moment.

Comment (falkreon) [reply build.gradle:47]: any "@Nonnull" requires JSR. There is a nonnull in the spec now.
Benefits:
* Better Kotlin integration
* "CLASS" retention type requires the JVM to not load it at runtime, so it's literally zero-weight.

**Review by LemmaEOF — APPROVED** [review]

File: src/main/java/io/github/cottonmc/energy/api/EnergyStorage.java [hunk 2]
```diff
@@ -1,48 +1,20 @@
...
-
-    /**
-     * Generate an amount of power, without needing to worry about packet sizes.
-     * DO NOT OVERRIDE WITHOUT CALLING THE EVENT HANDLER. DOING SO CAN AND WILL BREAK THINGS.
-     *
-     * @param amount The amount of energy to generate.
-     */
-    default void generateEnergy(int amount) {
-        if (canGenerateEnergy(amount)) {
-            for (Consumer<BlockEntity> handler : ((HandlerArray<Consumer<BlockEntity>>) PowerGenEvent.GENERATE_POWER).getBackingArray()) {
```
Comment (LemmaEOF) [inline src/main/java/io/github/cottonmc/energy/api/EnergyStorage.java:43]: Has this event call been moved anywhere yet? I do want to keep it in just so mods don't have to go through a super-contrived process to hook into generation events for stuff like stat tracking or pollution.

**Review by Pannoniae — APPROVED** [review]
I don't see any partial delivery in the PR but seems OK for me.

**Pull request merged**